src/query/phrase_query/regex_phrase_weight.rs

```diff
@@ -0,0 +1,487 @@
+use std::sync::Arc;
+
+use common::BitSet;
+use tantivy_fst::Regex;
+
+use super::PhraseScorer;
+use crate::fieldnorm::FieldNormReader;
+use crate::index::SegmentReader;
+use crate::postings::{LoadedPostings, Postings, SegmentPostings, TermInfo};
+use crate::query::bm25::Bm25Weight;
+use crate::query::explanation::does_not_match;
+use crate::query::union::{BitSetPostingUnion, SimpleUnion};
+use crate::query::{AutomatonWeight, BitSetDocSet, EmptyScorer, Explanation, Scorer, Weight};
+use crate::schema::{Field, IndexRecordOption};
+use crate::{DocId, DocSet, InvertedIndexReader, Score};
+
+type UnionType = SimpleUnion<Box<dyn Postings + 'static>>;
+
+/// The `RegexPhraseWeight` is the weight associated to a regex phrase query.
+/// See RegexPhraseWeight::get_union_from_term_infos for some design decisions.
+pub struct RegexPhraseWeight {
+    field: Field,
+    phrase_terms: Vec<(usize, String)>,
+    similarity_weight_opt: Option<Bm25Weight>,
+    slop: u32,
+    max_expansions: u32,
+    /// wildcard_mode is true if the query is interpeted as wildcard query instead of regex.
+    /// e.g. wol*
+    wildcard_mode: bool,
```

Comment:
nitpick: I think wildcard mode is an anti feature.

Reply:
true, I'll move the wildcard conversion to an utility function